mmengine/visualization/visualizer.py

```diff
@@ -194,12 +188,30 @@ def __init__(
             if None not in names and len(set(names)) != len(names):
                 raise RuntimeError('The name fields cannot be the same')
 
-            save_dir = osp.join(save_dir, 'vis_data')
+            # check if vis_backends are initialized
+            if save_dir is None:
+                for vis_backend in vis_backends:
+                    if vis_backend.get('save_dir') is None:
+                        print_log(
+                            '`Visualizer` backend is not initialized '
+                            'because save_dir is None '
+                            'and at least one vis_backend has no save_dir.',
+                            logger='current',
+                            level=logging.WARNING)
+                        break
 
-            for vis_backend in vis_backends:
-                name = vis_backend.pop('name', vis_backend['type'])
-                vis_backend.setdefault('save_dir', save_dir)
-                self._vis_backends[name] = VISBACKENDS.build(vis_backend)
+            else:
+                save_dir = osp.join(save_dir, 'vis_data')
+                for vis_backend in vis_backends:
+                    name = vis_backend.pop('name', vis_backend['type'])
+                    vis_backend.setdefault('save_dir', save_dir)
+                    self._vis_backends[name] = VISBACKENDS.build(vis_backend)
+        else:
+            print_log(
+                '`Visualizer` backend is not initialized '
+                'because vis_backends is None.',
+                logger='current',
+                level=logging.WARNING)
```

Comment:
```suggestion
```

Reply:
There is a test case in https://github.com/open-mmlab/mmengine/blob/main/tests/test_visualizer/test_visualizer.py#L74-L75

```python
        with self.assertLogs(MMLogger.get_current_instance(), level='WARNING'):
            Visualizer()
```

Delete else branch will fail the test

Reply:
We can remove this test.